assets/src/components/swingsView.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { ReactElement, useContext, useState } from "react"
+import { useRoutes } from "../contexts/routesContext"
+import { SocketContext } from "../contexts/socketContext"
+import { StateDispatchContext } from "../contexts/stateDispatchContext"
+import { isVehicle } from "../models/vehicle"
+import Loading from "./loading"
+import CloseButton from "./closeButton"
+import { flatten, uniq } from "../helpers/array"
+import { ghostIcon, upRightIcon } from "../helpers/icon"
+import { runIdToLabel } from "../helpers/vehicleLabel"
+import useSwings from "../hooks/useSwings"
+import useVehiclesForRunIds from "../hooks/useVehiclesForRunIds"
+import { ByRunId, VehicleOrGhost } from "../realtime"
+import { ByRouteId, Route, Swing } from "../schedule"
+import { selectVehicle, toggleSwingsView } from "../state"
+import {
+  formattedScheduledTime,
+  now,
+  serviceDaySeconds,
+} from "../util/dateTime"
+
+const SwingsView = (): ReactElement<HTMLElement> => {
+  const [, dispatch] = useContext(StateDispatchContext)
+  const [currentTime] = useState(now())
+  const swings = useSwings()
+
+  const activeSwings = swings
+    ? swings.filter((swing) => {
+        return swing.time > serviceDaySeconds(currentTime)
+      })
+    : []
+
+  const swingRunIds = uniq(
+    flatten(activeSwings.map((swing) => [swing.fromRunId, swing.toRunId]))
+  )
+
+  const { socket } = useContext(SocketContext)
+  const swingVehicles = useVehiclesForRunIds(socket, swingRunIds)
+  let swingVehiclesByRunId: ByRunId<VehicleOrGhost> = {}
```

Comment:
we can eliminate this `let` here (which is almost always a good idea) by replacing the `if` with a ternary operator, turning it into an expression